### ISSUE TITLE
Update Google.Cloud.Logging.Console with GA Microsoft.Extensions.Logging.Console

### ISSUE THREAD

**Issue by jskeet** [googleapis/google-cloud-dotnet#5578]
Now Microsoft.Extensions.Logging.Console is GA, we can productionize our logging formatter.

**Comment by umitkavala** [comment]
I'll start working: 

- [ ] Updating dependencies
- [ ] Creating an integration test
- [ ] Productionizing the "currently somewhat prototype" code - and particularly thinking about performance
- [ ] Writing more unit tests
- [ ] Writing documentation

**Comment by umitkavala** [comment]
Are we going to increase version in apis.json?
```
"id": "Google.Cloud.Logging.Console",
 "version": "1.0.0-alpha00",
```

**Comment by jskeet** [comment]
Not until it's ready to release

**Comment by umitkavala** [comment]
I've did quick experiment for performance. 
### **Default Implementation** 
![image](https://user-images.githubusercontent.com/3861934/100389911-0c20bb80-302f-11eb-9912-d9ddbcccc722.png)


### **With System.IO.Pipelines** > https://devblogs.microsoft.com/dotnet/system-io-pipelines-high-performance-io-in-net/

![image](https://user-images.githubusercontent.com/3861934/100389814-b3512300-302e-11eb-989f-39aced37da64.png)

There is no difference on allocations. Maybe I miss something.

![image](https://user-images.githubusercontent.com/3861934/100390024-5dc94600-302f-11eb-984a-30ff702f451b.png)

### Implementation
![image](https://user-images.githubusercontent.com/3861934/100390208-da5c2480-302f-11eb-9119-34f4b698666f.png)

i'll also try with **RecyclableMemoryStream** 

Do I need to create new issue for this?

**Comment by jskeet** [comment]
No need to create an issue for it particularly. I won't be able to review anything until I'm back at work on Monday though.

**Comment by umitkavala** [comment]
@jskeet I've updates benchmark test results with proper testing. (above)

**Comment by jskeet** [comment]
Thanks. It's still frustrating to me that we have to create a string at all. But I haven't looked at alternatives yet. (Perhaps try using `TextWriter.Write(ReadOnlySpan<char>)`?)

**Comment by umitkavala** [comment]
I'll try to avoid string 👍 and try ReadOnlySpan

**Comment by umitkavala** [comment]
Actually System.IO.Pipe has `ReadOnlySequence<byte>` to write textWriter. 
```c#
var reader = pipe.Reader;
reader.TryRead(out var result);
textWriter.WriteLine(result.Buffer);
```

**Comment by umitkavala** [comment]
![image](https://user-images.githubusercontent.com/3861934/101408933-66e2ce80-38dd-11eb-837a-f82980b7a066.png)
All 3 implementation together (fastest to slowest):
**1- Default Implementation using  `TextWriter.Write(ReadOnlySpan<char>)` ( .NET Standard 2.1)** 
```c#
ReadOnlySpan<char> readOnlySpan = 
             new ReadOnlySpan<char>(Encoding.UTF8.GetChars(output.GetBuffer(), 0, (int)output.Position));
 textWriter.Write(readOnlySpan);
```
**2- System.IO.Pipelines** 
```c#
var reader = pipe.Reader;
reader.TryRead(out var result);
textWriter.WriteLine(result.Buffer);
```
**3- RecyclableMemoryStream +  `TextWriter.Write(ReadOnlySpan<char>)` ( .NET Standard 2.1)** 
Implementation almost same with default one except couple of setting value for buffer size etc..

**

> Which metric(s) are important to decide? speed, allocation?
> Is it ok to create draft PR for default implementation (1)
> We need to upgrade .net Standard 2.0 to 2.1 in order to user `ReadOnlySpan<char>` for write

**

**Comment by jskeet** [comment]
I'm surprised that there's still so much allocation, and I'd like to look at that more - but I'm afraid I won't be able to spend any time reviewing anything around this until the New Year.

**Comment by umitkavala** [comment]
Sure. In the mean time I'll have a look better solution. 
last thing : what would you focus on this issue? utfjsonwriter also has string operations.

I wish happy new year to you and your family.

**Comment by jskeet** [comment]
I'm afraid I'd need to spend longer looking at it than I have spare time right now just to work out what to focus on... I suggest you just explore the space.

**Comment by NinoFloris** [comment]
Stumbled onto this issue, is this still something that will be released for 5.0?

**Comment by jskeet** [comment]
@NinoFloris I'm not sure what you mean by "for 5.0" - 5.0 of what package, exactly? I would expect us to release a 1.0.0 of Google.Cloud.Logging.Console when we've found time to finish it.

**Comment by NinoFloris** [comment]
Right, yeah that would be .net 5.0, considering 6.0 is already running first previews. Initially you noted wanting to wait for 5.0 to be released before productionizing this. Rephrased my question was more, is this tied to the next (asp).net release, or just when ready. When ready it is :)

**Comment by jskeet** [comment]
It doesn't need to target *.NET* 5.0 or 6.0 - it's only `Microsoft.Extensions.Logging.Console` that I needed to go GA... and we can take a dependency on that while still only keeping to netstandard2.0.

Basically we're not waiting for anything else to be released - we're just waiting for it to be the most important way of spending our limited time.

**Comment by NinoFloris** [comment]
I understand, I really just reference these versions because Microsoft *does* release these things in lockstep, features for extension packages rarely get an accelerated timeline.

Either way, thanks for the heads-up, I'll manage with the existing google cloud logger for now!
